docs/devel/releasing.md

```diff
@@ -1,7 +1,156 @@
 # Releasing Kubernetes
 
-This document explains how to create a Kubernetes release (as in version) and
-how the version information gets embedded into the built binaries.
+This document explains how to cut a release, and the theory behind it. If you
+just want to cut a release and move on with your life, you can stop reading
+after the first section.
+
+## How to cut a Kubernetes release
+
+Regardless of whether you are cutting a major or minor version, cutting a
+release breaks down into four pieces:
+
+1. Selecting release components.
+1. Tagging and merging the release in Git.
+1. Building and pushing the binaries.
+1. Writing release notes.
+
+You should progress in this strict order without interleaving pieces of each
+major step unless you are very familiar with the release process.
```

Comment:
I'd rather we not allow "unless you are very familiar with the release process" -- individual discretion has resulted in semi-borked releases that were off or wrong in various ways.  It seems unlikely that any perceived benefits outweigh costs/risks for a v1+ project.